### PR TITLE
Fix respawn-at-anchor setting.

### LINKED ISSUE #5824: If "respawn-at-home-bed" and "respawn-at-home" are set to true in the EssentialsX config, "respawn-at-anchor" is ultimately ignored.
### Type of bug

Other unexpected behaviour

### `/ess dump all` output

N/A

### Error log (if applicable)

_No response_

### Bug description

The bugged code is in the `onPlayerRespawn` method in `EssentialsSpawnPlayerListener`. 

The `respawn-at-anchor` option works by simply returning without modifying the spawn location if the default Minecraft spawn is a respawn anchor and the `respawn-at-anchor` option is set to true. However, if it isn't set to true, while `respawn-at-home` and `respawn-at-home-bed` are, the respawn point will ultimately get set to the value of the method `getBedSpawnLocation()` in the Spigot API. In spite of this method's name, it doesn't just get bed spawn locations, but also gets the location of respawn anchors. 

As a result, if a player's respawn point is set to a respawn anchor and `respawn-at-home-bed` and `respawn-at-home` are true, the player will still respawn at the respawn anchor.

### Steps to reproduce

1. Set `respawn-at-anchor` to false, and `respawn-at-home` and `respawn-at-home-bed` to true in the Essentials configuration.
2. Mark your spawn at a respawn anchor.
3. Try respawning.

### Expected behaviour

Players should be unable to respawn at respawn anchors unless `respawn-at-anchor` is true.

### Actual behaviour

Players can respawn at respawn anchors if `respawn-at-home-bed` and `respawn-at-home` are set to true.

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by gibsonpil**
Closes #5824.

**Environments tested:**
Operating System: Debian 12
Java Version: `openjdk 21.0.3 2024-04-16 LTS`

- [x] Paper 1.20.6 latest
- [x] Spigot 1.12.2
- [x] Spigot 1.8.8

### CODE REVIEW

File: EssentialsSpawn/src/main/java/com/earth2me/essentials/spawn/EssentialsSpawnPlayerListener.java
Comment (gibsonpil) [inline EssentialsSpawn/src/main/java/com/earth2me/essentials/spawn/EssentialsSpawnPlayerListener.java:1]: I'm curious to know what the standard manner of handling recently deprecated functions without functional equivalents in older version is within EssentialsX. This is the best I could come up with, but if there is a preferred way of doing this I'd be happy to switch over to it.

Comment (abulujayn) [reply EssentialsSpawn/src/main/java/com/earth2me/essentials/spawn/EssentialsSpawnPlayerListener.java:1]: I believe the best way to do this is by using providers. https://github.com/EssentialsX/Essentials/tree/2.x/providers

You can create a base provider interface, then implement a legacy and modern one.

File: EssentialsSpawn/src/main/java/com/earth2me/essentials/spawn/EssentialsSpawnPlayerListener.java [hunk 1]
```diff
@@ -40,21 +40,23 @@ void onPlayerRespawn(final PlayerRespawnEvent event) {
...
+            if (ess.getSettings().isRespawnAtBed() &&
+                    (!VersionUtil.getServerBukkitVersion().isHigherThanOrEqualTo(VersionUtil.v1_16_1_R01) ||
+                    (!event.isAnchorSpawn() || ess.getSettings().isRespawnAtAnchor()))) {
                 // cannot nuke this sync load due to the event being sync so it would hand either way
-                bed = user.getBase().getBedSpawnLocation();
+                try {
+                    respawnLocation = user.getBase().getRespawnLocation();
+                } catch (NoSuchMethodError e) { // For versions prior to 1.16.
+                    respawnLocation = user.getBase().getBedSpawnLocation();
+                }
```
Comment (JRoy) [inline EssentialsSpawn/src/main/java/com/earth2me/essentials/spawn/EssentialsSpawnPlayerListener.java]: Please use `VersionUtil` checks here, that stack trace can be expensive especially if this is hot code

**Review by JRoy — CHANGES_REQUESTED** [review]

**Review by JRoy — CHANGES_REQUESTED** [review]
just one thing otherwise fine

**Comment by gibsonpil** [comment]
Alright, I switched it over to using `VersionUtil` and rebased everything. :smile:

File: EssentialsSpawn/src/main/java/com/earth2me/essentials/spawn/EssentialsSpawnPlayerListener.java
Comment (gibsonpil) [reply EssentialsSpawn/src/main/java/com/earth2me/essentials/spawn/EssentialsSpawnPlayerListener.java:1]: I looked over the other providers that were implemented and it appears that the bulk of them are for translating multiple class fields across different iterations of the Bukkit API. I'm not sure that it'd be worth it to implement such a thing for a one-off method call, and it'd certainly add unnecessary design complexity.

**Review by JRoy — APPROVED** [review]

**Comment by JRoy** [comment]
Thanks for the contribution :)

**Pull request merged**